pkg/agent/run.go

```diff
@@ -177,7 +177,7 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 		return errors.Wrap(err, "failed to wait for apiserver ready")
 	}
 
-	coreClient, err := util.GetClientSet(nodeConfig.AgentConfig.KubeConfigKubelet)
+	coreClient, err := util.GetClientSet(nodeConfig.AgentConfig.KubeConfigK3sController)
```

Comment:
We probably need to break this up. The configureNode stuff should use the kubelet RBAC to update itself. Other things (flannel/netpol) should use the controller RBAC without any permissions to update nodes.